salt/edx/prod.sls

```diff
@@ -31,26 +44,16 @@ configure_python_ppa_for_edx:
         - pkg: install_os_packages
 {% endif %}
 
-install_os_packages:
+{% for os_pkg in os_packages %}
+install_{{ os_pkg }}:
```

Comment:
There's no need to turn this into a for loop, you just replace the list parameter of the `pkgs` attribute with the jinja variable.
```
install_os_package:
  pkg.installed:
    - pkgs: {{ os_packages }}
```